src/render.rs

```diff
@@ -0,0 +1,135 @@
+//! Rendering is RLBot's ability to draw directly inside the game window.
+
+use flat;
+use flatbuffers::{FlatBufferBuilder, WIPOffset};
+use rlbot::RLBot;
+use std::error::Error;
+
+/// A render group in the process of being built.
+///
+/// The drawing methods in this class simply collect a list of items to be
+/// drawn later. Nothing will actually be drawn to screen until
+/// [`render`](RenderGroup::render) is called.
+///
+/// # Examples
+///
+/// ## Basic rendering
+///
+/// ```no_run
+/// # use rlbot::flat::ColorArgs;
+/// # use std::error::Error;
+/// #
+/// # fn main() -> Result<(), Box<Error>> {
+/// let rlbot = rlbot::init()?;
+/// let mut group = rlbot.begin_render_group(1234);
+/// group.draw_string_2d(10.0, 10.0, 2, 2, "I am text!", &ColorArgs::rgb(0, 255, 0));
+/// group.render()?;
+/// # Ok(())
+/// # }
+/// ```
+///
+/// ## Clearing the screen
+///
+/// ```no_run
+/// # use std::error::Error;
+/// #
+/// # fn main() -> Result<(), Box<Error>> {
+/// let rlbot = rlbot::init()?;
+/// let group = rlbot.begin_render_group(1234);
+/// group.render()?;
+/// # Ok(())
+/// # }
+/// ```
+pub struct RenderGroup<'a> {
+    rlbot: &'a RLBot,
```

Comment:
I guess a consequence of this is that the user has to make sure that the `RenderGroup` goes out of scope before they can use `rlbot` again, right? And you can't have two active `RenderGroup`s at the same time.

Not the end of the world, but given that only the `render` call needs this, might make sense to make `render` just take a `&RLBot`, so the borrow is limited to the minimum scope.